src/simularium/VisGeometry.ts

```diff
@@ -1105,34 +1135,28 @@ class VisGeometry {
      */
     private mapIdToGeom(
         id: number,
-        meshName: string | undefined,
-        pdbName: string | undefined,
-        assetPath: string
+        displayType: string,
+        url?: string,
+        color?: string
     ): void {
-        this.logger.debug(`Mesh for id ${id} set to '${meshName}'`);
-        this.logger.debug(`PDB for id ${id} set to '${pdbName}'`);
+        this.logger.debug(`Geo for id ${id} set to '${url}'`);
         const unassignedName = `${VisAgent.UNASSIGNED_NAME_PREFIX}-${id}`;
+        const isMesh = displayType === "OBJ";
+        const isPBD = displayType === "PDB";
+        console.log(color); // TODO: handle color
+        if (!url) {
+            // displayType not either pdb or obj, will show a sphere
+            // TODO: handle CUBE, GIZMO etc
+            return;
+        }
         this.visGeomMap.set(id, {
-            meshName: meshName || DEFAULT_MESH_NAME,
-            pdbName: pdbName || "",
+            meshName: isMesh ? url : DEFAULT_MESH_NAME,
+            pdbName: isPBD ? url : "",
```

Comment:
if mesh and pdb are mutually exclusive now, it seems to me the visGeomMap could just have one single object of displayType and url.  Not necessarily for this pr but I think it would work.

Reply:
yeah, I think the whole, load, keep track of requests, and save the geo data could be its own class, and yeah, I think this sort of change would belong in that set